source/develop/release-management/features/network/openstack_identity_api_v3.html.md

```diff
@@ -0,0 +1,290 @@
+* Tokens explicitly represent user+project or user+domain pairs
+* The token is returned the X-Subject-Token response header
+
+A typical request to get a token in OpenStack Identity API v2.0 is a HTTP `POST`
+to `/v3/auth/tokens` with the content:
+~~~~
+{
+  "auth" : {
+    "identity" : {
+      "methods" : [ "password" ],
+      "password" : {
+        "user" : {
+          "domain" : {
+            "name" : "Default"
+          },
+          "name" : "admin",
+          "password" : "c314d360aab94a5f"
+        }
+      }
+    },
+    "scope" : {
+      "project" : {
+        "domain" : {
+          "name" : "Default"
+        },
+        "name" : "admin"
+      }
+    }
+  }
+}
+~~~~
+The `scope` might be not required, e.g. if a default project is configured for
+the user. The relevant part of the response looks like this:
+~~~~
+HTTP/1.1 201 Created
+X-Subject-Token: gAAAAABbH3sSGLgtxvoHyYDLZd6fSuserBHG2P-3TdykaYnq-eYo7aUBdZK0I
+Vary: X-Auth-Token
+Content-Type: application/json
+
+{"token": ... }
+~~~~
+
+### Implementation in oVirt Engine
+
+To support OpenStack Identity API v3 oVirt engine has to be extended to handle
+the user domain name, the project name and the domain name of the project.
+This requires changes in webadmin, REST-API, the database and the 
+ExternalNetworkTokenProvider in the backend.
+
+The version of the OpenStack Identity API is derived from the authentication
+URL. If the authentication URL ends with "/v3" or "/v3/", OpenStack Identity
+API v3 will be used, else OpenStack Identity API v2.0 to ensure backward
+compatibility.
+
+The default ovirt-provider-ovn has not be modified, because the OpenStack
+Identity API v2.0 is not removed.
+
+### Implementation in oVirt Provider OVN
+To support OpenStack Identity API v3 the oVirt Provider OVN to be extended to
+handle requests to `/v3/auth/tokens`. The `scope` can be ignored, like the
+`tenantName`.
+In oVirt Provider OVN's OpenStack Identity API v2.0 implementation the
+user name has to match the following format:
+`<admin_username>[@<fqdn>]@<ovirt_profile>`, while the optional `<fqdn>` is
+the Active Directory or LDAP domain. For the reason of consistency, this format
+is kept in the OpenStack Identity API v3 implementation and the `user.domain`
+is ignored.
+
+## Prerequisites
+
+oVirt Engine requires an updated version of [OpenStack Java SDK 3.2.x][9] to
+accessing OpenStack Identity API v3.
+
+## Limitations
+
+OpenStack Identity API provides a catalog of multiple endpoints, which
+provides the URL to access the OpenStack service. This feature does change
+the current implementation in oVirt Engine, which requires the user to specify
+this URL manually.
+
+## Benefit to oVirt
+
+Support for OpenStack Identity API v3 in oVirt Engine is required to enable 
+access to services from [OpenStack instances which does not support Identity
+API v2.0, especially the OpenStack releases since Qeens.][12]
+
```

Comment:
As i see now the new UI 'm wondering about 2 things:
- There should be than an option to choose v2 or v3 via the engine-setup when installing the ovn-provider for the first time. 
- In the UI i saw we have a 'DC' drop down list, why? isn't provider is a global engine thing? It's not a DC entity as i get it, or i'm wrong?

Reply:
> There should be than an option to choose v2 or v3 via the engine-setup when installing the ovn-provider for the first time.

We could have this, but why we should do this?

> In the UI i saw we have a 'DC' drop down list, why? isn't provider is a global engine thing? It's not a DC entity as i get it, or i'm wrong?

This seems to be something specific to block storage.

Reply:
Because v2 is deprecated, so why it's remain the default? user need to choose, now he have 2 option and not 1.

Block what exactly on storage? i don't understand why provider become a DC entity suddenly? how it effects us?

Reply:
I think this is not related to keystone v3 support.